### PR TITLE
Teach swift-ide-test to print type interface via a given mangled type name.

### PR DESCRIPTION
**Pull request by nkcsgexi**
[ide] Teach swift-ide-test to print type interface via a given mangled type name. related rdar://27306890

#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Comment by nkcsgexi** [comment]
@swift-ci please test

File: test/IDE/print_type_interface.swift [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// XFAIL: linux
```
Comment (jrose-apple) [inline test/IDE/print_type_interface.swift]: Why the Linux XFAIL?

Comment (nkcsgexi) [inline test/IDE/print_type_interface.swift]: No worries, I will remove it.

**Comment by nkcsgexi** [comment]
@swift-ci Please smoke test OS X platform

**Comment by nkcsgexi** [comment]
@swift-ci Please test

**Comment by nkcsgexi** [comment]
@swift-ci Please test

**Comment by nkcsgexi** [comment]
@swift-ci Please test

**Comment by nkcsgexi** [comment]
Finally all related checks passed. Happily merging.

**Pull request merged**